src/Data/LLVM/BitCode/Parse.hs

```diff
@@ -574,6 +583,33 @@ getTypeId n = do
 
 -- Value Symbol Table ----------------------------------------------------------
 
+-- | An LLVM Bitcode file, it is comprised of nested Blocks of information,
+-- with records available at each block.  Different blocks hold different
+-- types of information, and the nesting represents program scope and what is
+-- defined/accessible at a particular point within the program.
+--
+-- The `Parse` and `Finalize` monads maintain a set of symbol tables to use
+-- for lookups, as stored in the `Env` structure referenced by those monads:
+--
+--   * `valSymtab` for value references
+--
+--   * `fnSymtab` for function references
+--
+--   * `bbSymTab` for basic-block references, either "named" by association
+--     with a symbol name (like the function's entry block) or "anonymous",
+--     where there is no direct symbol but there may be a block label (like
+--     for a goto statement or similar surface language construct.)
+--
+-- These lookups are necessary to "relabel" values that reference symbol or
+-- label addresses in the code with the actual targets as resolved by
+-- processing the entirety of the LLVM bitcode file.  The
+-- llvm-pretty-bc-parser runs in two phases: the initial phase where it
+-- processes the LLVM bitcode stream to create "Partial" representations of
+-- all of the elements, followed by a "finalization" phase where it performs
+-- all of the label references above (via the llvm-pretty `relabel`
+-- operation), as well as other fixups to convert the "Partial" data
+-- structures into the structures defined by the `llvm-pretty` AST.
+
 type SymName = Either String Int
```

Comment:
Currently, these Haddocks are describing the `type SymName = Either String Int` type synonym, which doesn't seem like what you intended. Should these Haddocks describe a different definition? Perhaps this was meant to be a Haddock named section (e.g., `$llvmBitcodeFile`) in the export list? Something else?